stdlib/domain.mli

```diff
@@ -39,27 +39,25 @@ val self : unit -> id
 (** [self ()] is the identifier of the currently running domain *)
 
 val at_first_spawn : (unit -> unit) -> unit
-(** Register the given function to be called before any domain is
-    spawned (except the initial domain) and before any function registered
-    with {!val:at_startup} is called. The functions registered with
-    [at_first_spawn] are called in 'last in, first out' order: the function
-    most recently added with [at_first_spawn] is called first.
+(** [at_first_spawn f] calls [f] before the first domain is spawned by the
+    program. The functions registed with [at_first_spawn] are called on the
+    main (initial) domain. The functions registered with [at_first_spawn] are
+    called in 'first in, first out' order: the oldest function added with
+    [at_first_spawn] is called first.
 
-    @raise Invalid_argument if the first domain has already been spawned. *)
+    @raise Invalid_argument if the program has already spawned a domain. *)
+
+val at_each_spawn : (unit -> unit) -> unit
+(** [at_each_spawn f] calls [f] on the newly spawned domains. The function [f]
```

Comment:
Style/writing comment: I find the first sentence here confusing. Personally I read `[at_each_spawn f] calls [f]` as saying that `f` will be called immediately when this call is evaluated, so the precision "on the newly spawned domains" is weird. (It will be called right away, in parallel on a bunch of domains?) I'm not sure what formulation to suggest.

This comment somewhat applies to the documentation of the three functions that invariably starts with `[foo f] calls [f] ...`. Maybe a consistently different opening could be used. At the same time the previous implementation-focused style (`[foo f] registers [f] to be called when ...`) was maybe a bit too low-level. Do people have recommendations?

Note: the doc of the other two functions start as follows:
- `[at_exit f] calls [f] when the current domain exits.`
- `at_first_spawn f] calls [f] before the first domain is spawned by the program.`

Somehow I find them slightly less confusing; in particular`calls [f] when ...` is less confusing thanks to the "when" that makes it clear that `calls [f]` does not mean "right away". Still I would prefer a clearer formulation, hopefully that would work in the three cases.

Reply:
I'm going with `[foo f] registers [f] to be called`. This wording feels to be least likely to be misinterpreted.